Marlin/src/gcode/calibrate/G28.cpp

```diff
@@ -408,17 +408,37 @@ void GcodeSuite::G28() {
       #endif
     #endif
 
+    // Z may home first, e.g., when homing away from the bed
     TERN_(HOME_Z_FIRST, if (doZ) homeaxis(Z_AXIS));
 
+    // 'R' to specify a specific raise. 'R0' indicates no raise, e.g., for recovery.resume
+    // When 'R0' is used, there should already be adequate clearance, e.g., from homing Z to max.
     const bool seenR = parser.seenval('R');
-    const float z_homing_height = seenR ? parser.value_linear_units() : Z_HOMING_HEIGHT;
 
-    if (z_homing_height && (seenR || NUM_AXIS_GANG(doX, || doY, || TERN0(Z_SAFE_HOMING, doZ), || doI, || doJ, || doK, || doU, || doV, || doW))) {
-      // Raise Z before homing any other axes and z is not already high enough (never lower z)
-      if (DEBUGGING(LEVELING)) DEBUG_ECHOLNPGM("Raise Z (before homing) by ", z_homing_height);
-      do_z_clearance(z_homing_height);
-      TERN_(BLTOUCH, bltouch.init());
+    // Use raise given by 'R' or Z_HOMING_HEIGHT (above the probe trigger point)
+    float z_homing_height = seenR ? parser.value_linear_units() : DIFF_TERN(HOMING_Z_WITH_PROBE, Z_HOMING_HEIGHT, _MIN(probe.offset.z, 0));
```

Comment:
I changed this to `DIFF_TERN` rather than `SUM_TERN` because it makes more sense (to me) that we would do more of a raise in the case where `probe.offset.z` is farther from the bed. But again, the inclusion of `probe.offset.z` here is somewhat dubious, since the configured `Z_HOMING_HEIGHT` is already supposed to be accounting for that. That is, it should be defined by the user as larger than the largest `-probe.offset.z` they expect to have.

Reply:
Yeah, swapping to absolute moves is fine, so `probe.offset.z` will be uneccesary in many places.